Lib/test/test_configparser.py

```diff
@@ -720,6 +721,11 @@ def test_read_returns_file_list(self):
         parsed_files = cf.read(file1)
         self.assertEqual(parsed_files, [file1])
         self.assertEqual(cf.get("Foo Bar", "foo"), "newbar")
+        # check when we pass only a Path object:
+        cf = self.newconfig()
+        parsed_files = cf.read(pathlib.Path(file1))
+        self.assertEqual(parsed_files, [pathlib.Path(file1)])
```

Comment:
At the moment the behaviour is consistent with the current behaviour if the Path object was passed as part of a list. Making it return a string would involve changing this behaviour too if we're being consistent. We would also have to be careful not to break things if anybody's passing bytes paths in as part of a list.

I'm not certain it makes as much sense here as it does in zipfiles as the config file doesn't contain the same kind of internal structure, are there other examples where it's being converted?